widget/select.go

```diff
@@ -235,13 +235,12 @@ func (s *Select) popUpPos() fyne.Position {
 }
 
 func (s *Select) showPopUp() {
-	var items []*fyne.MenuItem
-	for _, option := range s.Options {
+	items := make([]*fyne.MenuItem, len(s.Options))
+	for i, option := range s.Options {
```

Comment:
Maybe this could be re-written? 🤔:
```go
for i := range s.Options {
     text := s.Options[i] // capture
     items[i] = fyne.NewMenuItem(text, func() {
	 s.optionTapped(text)
      })
}
```

Reply:
It could, but is there any benefit to doing so?

Reply:
yes but it is almost imperceptible I think. The benefit is that `option` variable will not be created once and will not be assigned on every iteration without any actual need

Reply:
I thought that the compiler wouldn't be smart enough to optimize that away (I thought it would update the range value each time anyway), but you are right. The compiler is smart enough and there is a very small difference 🙂

Reply:
To avoid any doubt about range 😄:
```go
for i := 0; i < len(s.Options); i++ {
     text := s.Options[i] // capture
     items[i] = fyne.NewMenuItem(text, func() {
	 s.optionTapped(text)
      })
}
```

Reply:
Yeah. I will stick with the first for now. https://github.com/fyne-io/fyne/pull/2181 will drop this code out of existence.